src/app/youtube/services/detailed-information.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { SearchItem } from '../models/search-item.model';
+import { SearchOptionsService } from 'src/app/shared/services/search-options.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DetailedInformationService {
+
+    public selectedItem: SearchItem = null;
+
+    public datesLag: number = null;
+
+    constructor(private searchOptionsService: SearchOptionsService) { }
+
+    public initDetailedInformation(clickedItem: SearchItem): void {
+        this.selectedItem = clickedItem;
+        this.initDatesLag(clickedItem);
+        this.searchOptionsService.disableInput();
+    }
+
+    public initDatesLag(clickedItem: SearchItem): void {
+        const today: Date = new Date();
+        this.datesLag =
+            ((today.getTime() - new Date(clickedItem.snippet.publishedAt).getTime()) / (1000 * 3600 * 24));
```

Comment:
I saw such function before in your code. It is a good point to create separate helper function